opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -175,7 +174,7 @@ impl<R: RuntimeChannel> LogProcessor for BatchLogProcessor<R> {
                 name: "batch_log_processor_emit_error",
                 error = format!("{:?}", err)
             );
-            global::handle_error(LogError::Other(err.into()));
+            otel_error!(name: "BatchLogProcessor.emit", otel_error = "BatchLogProcessor.emit", error = format!("{:?}", LogError::Other(err.into())));
```

Comment:
Need to re-visit this, as this can flood if channel is full.

Reply:
This needs to be discussed whether it's necessary to handle these errors internally by implementing logic to prevent log flooding. This could involve using an "already-sent" flag and tracking the last sent timestamp to reduce the frequency of repeated error messages. 
Or else, let the user handle this flooding by adding a filter logic.

Reply:
> Or else, let the user handle this flooding by adding a filter logic.

OTel SDK, by default, should not burden the user with such requirements!
Few thoughts:
1. If the item is being dropped due to buffer being full, emit a log only once in the app lifetime. If item is dropped for other unknown reasons, it maybe fine to emit log for each occurrence. I prefer that this should be a simple bool flag only and not a sophisticated throttling mechanism.
2. We should use an internal metric for items dropped. This may require more refactoring/designs.
3. Keep the count of items dropped (using atomic uint etc.). During shutdown/drop, if that count is greater than zero, emit a warning log "N items were dropped due to buffer full".

All of these are good for future additions, so no need to tackle in this PR. 
My only minimum bar for RC release - do not log for each time item is dropped due to buffer is full. (We had similar issue for Metrics overflow that was fixed recently)